chore(web): blog overhaul

icepuma opened the pull request:
- make blog work again -> use new content layer stuff
- prepare blog series
- make sure drafts are not shown in featured list

rawkode commented:
I can't run this locally:

```
Cannot destructure property 'type' of 'lookupMap[collection]' as it is undefined.
  Location:
    /home/rawkode/Code/src/github.com/RawkodeAcademy/RawkodeAcademy/projects/rawkode.academy/platform/web/node_modules/astro/dist/content/runtime.js:487:17
  Stack trace:
    at Object.eval [as transform] (/home/rawkode/Code/src/github.com/RawkodeAcademy/RawkodeAcademy/projects/rawkode.academy/platform/web/node_modules/astro/dist/content/runtime.js:487:17)
    at async ZodObject.safeParseAsync (file:///home/rawkode/Code/src/github.com/RawkodeAcademy/RawkodeAcademy/projects/rawkode.academy/platform/web/node_modules/zod/lib/index.mjs:731:24)
    at async parseData (file:///home/rawkode/Code/src/github.com/RawkodeAcademy/RawkodeAcademy/projects/rawkode.academy/platform/web/node_modules/astro/dist/content/content-layer.js:148:40)
    at async eval (/home/rawkode/Code/src/github.com/RawkodeAcademy/RawkodeAcademy/projects/rawkode.academy/platform/web/node_modules/astro/dist/content/loaders/glob.js:192:13)
```

icepuma commented:
I fixed the `bun.lockb` file. I removed `.astro`, `dist`, `node_modules` and re-ran `bun install` and `ZITADEL_URL="aaa" ZITADEL_CLIENT_ID="bbb" bun run build` which works flawlessly :)

rawkode commented on projects/rawkode.academy/platform/web/.npmrc:
```diff
@@ -0,0 +1 @@
+registry=https://registry.npmjs.org
```
Why do we need this?

icepuma replied:
Yes! When you have a system-wide config which overrides this, the content of `bun.lockb` changes

rawkode replied:
Did you have a system wide config? 😛

rawkode commented on projects/rawkode.academy/platform/web/package.json:
```diff
@@ -18,7 +18,7 @@
     "@astrojs/sitemap": "3.2.1",
     "@astrojs/tailwind": "5.1.3",
-    "@astrojs/ts-plugin": "^1.10.4",
+    "@astrojs/ts-plugin": "1.10.4",
```
Do we need to pin this to a specific patch release?

icepuma replied:
This should not be a problem. When you run `bunx npm-check-updates --interactive` you'll get all the updates :)

rawkode commented on projects/rawkode.academy/platform/web/src/content.config.ts:
```diff
@@ -0,0 +1,33 @@
+  }),
+  schema: z.object({
+    title: z.string(),
+    series: reference("series").optional(),
```
This is cool
